### PR TITLE
Clean and fix default facet patch id map

### PR DESCRIPTION
**Pull request by sloriot**
The current code was breaking some user code

### CODE REVIEW

**Comment by maxGimeno** [comment]
Warning: https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-59/Polyhedron_Demo/TestReport_lrineau_ArchLinux-CXX17-Release.gz

File: Mesh_3/include/CGAL/Mesh_3/experimental/Facet_patch_id_map.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+// Primitive::Id is a std::pair
+template <typename MeshDomain,
+          typename Primitive>
+struct Facet_patch_id_map<MeshDomain, Primitive, false>
+{
+  typedef typename MeshDomain::AABB_primitive::Id Id;
+  typedef typename MeshDomain::Patch_id value_type;
+  typedef const value_type& reference;
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_3/experimental/Facet_patch_id_map.h:57]: The property map returns a value, and not a reference. That is why its `reference` type must be identical to `value_type`.
```suggestion
  typedef value_type reference;
```

`reference` is required to be an actual reference type only for models of [`LvaluePropertyMap`](https://www.boost.org/doc/libs/1_75_0/libs/property_map/doc/LvaluePropertyMap.html).

**Comment by maxGimeno** [comment]
There is one runtime error [here](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-60/Mesh_3/TestReport_lrineau_Ubuntu-latest-GCC6-CXX1z.gz), is it related ?

**Comment by lrineau** [comment]
> There is one runtime error [here](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-60/Mesh_3/TestReport_lrineau_Ubuntu-latest-GCC6-CXX1z.gz), is it related ?

The assertion `Assertion `volume >= volume_min'` is a well-known issue in CGAL testsuite. Probably the `volume_min` value is a bit too tight, and sometimes the produced meshed object has a lower volume.

**Comment by maxGimeno** [comment]
Ok, that's what I thought. Thank you !

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-65.shtml

**Pull request merged**